Libraries/Network/XMLHttpRequest.js

```diff
@@ -193,26 +193,25 @@ class XMLHttpRequest extends EventTarget(...XHR_EVENTS) {
       return this._cachedResponse;
     }
 
-    switch (this._responseType) {
+    switch (responseType) {
       case 'document':
         this._cachedResponse = null;
         break;
 
       case 'arraybuffer':
-        this._cachedResponse = toArrayBuffer(
-          this._responseText, this.getResponseHeader('content-type') || '');
+        this._cachedResponse = base64.toByteArray(this._response).buffer;
```

Comment:
Pretty much, with the added twist of trying to look at the Content-Type header as well (which shouldn't matter if I want bytes -- the exact bytes that came down the pipe).

The way I see it, without an API to create ArrayBuffers inside JSC (which is only coming in iOS 10), we're left with encoding bytes as strings in some sort of manner. One option would be `latin-1` as an encoding, mapping bytes 0-255 to Unicode characters 0-255 (the Latin-1 range). The other option is base64 which introduces 33% overhead, but is ASCII only and consistent what we do in other places (e.g. WebSockets). It also feels a bit less hacky, but meh.

I don't care which approach we use. We can also change it later if we discover that base64 adds too much overhead.